pex/jobs.py

```diff
@@ -595,3 +614,195 @@ def spawned_jobs():
+    :param verb: A verb indicating what the function does; "process" by default.
+    :param verb_past: The past tense of `verb`; "processed" by default.
+    :return: An iterator over the mapped results.
+    """
+    input_items = list(inputs)
+    if not input_items:
+        return
+
+    if costing_function is not None:
+        # We ensure no job slot is so unlucky as to get all the biggest jobs and thus become an
+        # un-necessarily long pole by sorting based on cost. Some examples to illustrate the effect
+        # using 6 input items and 2 job slots:
+        #
+        # 1.) Random worst case ordering:
+        #         [9, 1, 1, 1, 1, 10] -> slot1[9] slot2[1, 1, 1, 1, 10]: 14 long pole
+        #     Sorted becomes:
+        #         [10, 9, 1, 1, 1, 1] -> slot1[10, 1, 1] slot2[9, 1, 1]: 12 long pole
+        # 2.) Random worst case ordering:
+        #         [6, 4, 3, 10, 1, 1] -> slot1[6, 10] slot2[4, 3, 1, 1]: 16 long pole
+        #     Sorted becomes:
+        #         [10, 6, 4, 3, 1, 1] -> slot1[10, 3] slot2[6, 4, 1, 1]: 13 long pole
+        #
+        input_items.sort(key=costing_function, reverse=True)
+
+    # We want each of the job slots above to process MULTIPROCESSING_MIN_AVERAGE_LOAD on average in
+    # order to overcome multiprocessing overheads. Of course, if there are fewer available cores
+    # than that or the user has pinned max jobs lower, we clamp to that. Finally, we always want at
+    # least two slots to ensure we process input items in parallel.
+    pool_size = max(2, min(len(input_items) // min_average_load, _sanitize_max_jobs(max_jobs)))
+
+    perform_install = functools.partial(_apply_function, function)
+
+    slots = defaultdict(list)  # type: DefaultDict[int, List[float]]
+    with TRACER.timed(
+        "Using {pool_size} parallel jobs to {verb} {count} items".format(
+            pool_size=pool_size, verb=verb, count=len(input_items)
+        )
+    ):
+        with _mp_pool(size=pool_size) as pool:
+            for pid, result, elapsed_secs in pool.imap_unordered(perform_install, input_items):
+                TRACER.log(
+                    "[{pid}] {verbed} {result} in {elapsed_secs:.2f}s".format(
+                        pid=pid,
+                        verbed=verb_past,
+                        result=result_render_function(result) if result_render_function else result,
+                        elapsed_secs=elapsed_secs,
+                    ),
+                    V=2,
+                )
+                yield result
+                slots[pid].append(elapsed_secs)
+
+    TRACER.log(
+        "Elapsed time per {verb} job:\n  {times}".format(
+            verb=verb,
+            times="\n  ".join(
+                "{index}) [{pid}] {total_secs:.2f}s {count} {wheels}".format(
+                    index=index,
+                    pid=pid,
+                    count=len(elapsed),
+                    wheels=pluralize(elapsed, "wheel"),
```

Comment:
Should this be the following?
```suggestion
                    items=pluralize(elapsed, "item"),
```

Reply:
Fixed, though by parametrizing a `noun` instead.